build/api-docs-generator.js

```diff
@@ -0,0 +1,655 @@
+  }
+}
+
+
+/**
+ * fix notification UI broken areas
+ * NotificationUI file has some issues and mdx is not being rendered properly. Fixing it.
+ * Removing a line where docusaurus throws error
+ * Removing </br> tag that's within div tag. Only two instances of this in whole project, so fixing it by searching for the exact line instead of writing a logic (as the logic may disturb content from other files)
+ * @param {string} filePath
+ */
+function fixNotificationUiFile(filePath) {
+  try {
+    // Read the file synchronously
+    let content = fs.readFileSync(filePath, 'utf8');
+
+    let updatedContent = content;
+
+    // Fix notification UI broken areas
+    updatedContent = updatedContent.replace('ame="badge badge--info margin-left--sm">static</span><br/><a href="#done">done</a><span className="badge badge--info margin-left--sm">static</span></div></div>\n</div>', '');
+
+    updatedContent = updatedContent.replace('Click me to </br>locate the file in file tree', 'Click me to locate the file in file tree');
+    
+    // Replace second occurrence
+    updatedContent = updatedContent.replace('Click me to </br>locate the file in file tree', 'Click me to locate the file in file tree');
+
+    // Write the file synchronously
+    fs.writeFileSync(filePath, updatedContent, 'utf8');
+
+    console.log("Successfully fixed all issues inside notificationUI file");
+  } catch (err) {
+    console.error('Error:', err);
+  }
+}
+
+fixEventManagerFile(eventManagerPath);
+fixNotificationUiFile(notificationUIPath);
+
+
+/**
+ * Moves all MDX files and subfolders to a specified destination folder.
+ * @param {string} sourceDir - The source directory containing MDX files and folders
+ * @param {string} destDir - The destination directory for API reference
+ */
+function moveToApiReference(sourceDir, destDir) {
+  if (!fs.existsSync(destDir)) {
+    fs.mkdirSync(destDir, { recursive: true });
+  }
+
+  const entries = fs.readdirSync(sourceDir, { withFileTypes: true });
+
+  for (const entry of entries) {
+    const srcPath = path.join(sourceDir, entry.name);
+    const destPath = path.join(destDir, entry.name);
+
+    if (entry.isDirectory()) {
+      // Skip the "API reference" folder itself
+      if (entry.name === "API reference") continue;
+      
+      // Create the corresponding directory in the destination
+      if (!fs.existsSync(destPath)) {
+        fs.mkdirSync(destPath, { recursive: true });
+      }
+      
+      // Move contents of the directory
+      moveToApiReference(srcPath, destPath);
+      
+      // Remove the now-empty directory from the source
+      if (fs.readdirSync(srcPath).length === 0) {
+        fs.rmdirSync(srcPath);
+      }
+    } else if (path.extname(entry.name) === '.mdx') {
+      // If it's an MDX file, move it
+      fs.renameSync(srcPath, destPath);
+    }
+  }
+}
+
+const apiReferenceDir = path.join(__dirname, 'api', 'API reference');
+moveToApiReference(path.join(__dirname, 'api'), apiReferenceDir);
```

Comment:
Can we move all the generated docs to `docs/generatedApiDocs` folder. delete the contents in the generated docs forlder and replace with the newly built contents. Can we just copy the contents of the docs folder in phoenix repo and paste it into the docs repo api folder and things will work?

Reply:
Sure, we can achieve this. 

Should we name the directory "API Reference" or "generatedApiDocs"? The directory name is what appears as the collapsible panel in the Docs site.

Reply:
Anything , up to you!